### DISCUSSION TITLE
Roadmap to 1.0 release

### DISCUSSION THREAD

**Discussion by nickw444** [nickw444/flask-ldap3-login#99] [General]
Hey @gmacon and @HeMan,

Just starting this thread to discuss roadmap toward 1.0 release, or some incremental release at this point. As we've realized, patching the recent bug in #97 wasn't able to be quickly released due to the current in-flight (and potentially untested/breaking changes on master).

In the interim I pushed a new branch (`release-0.9`) who's HEAD is the latest 0.9 release and have cherry-picked the bug fix onto it. 

I'm opening this discussion topic to decide how we move forward, and whether we should implement any changes to our branches in this repo. 

I'm considering:

* Move `master` to `develop`.
* Rename `master` to `main` to align with recent Github changes
* Set `HEAD` of `main` to the latest `0.9` release. 

This however means we lose a lot of the new `1.0` goodness we've got lately, namely linting, formatting and new CI pipeline. But perhaps that's more fuel toward the 1.0 fire?

**Comment by nickw444** [comment]
I've just pushed `1.0.0a1` based on current state of `master` @ f1f3a35

https://github.com/nickw444/flask-ldap3-login/releases/tag/1.0.0a1

**Comment by andrewjroth** [comment]
> we lose a lot of the new 1.0 goodness we've got lately

Could you explain why moving branches around would result in loss of functionality?

It's been almost two years since 1.0 was started.  What would it take to release it?  I'm happy to jump in and contribute, if needed.

**Comment by ArthurMitchell42** [comment]
Hi Nick,
I'm checking functionality on the master branch and I've just notice that V1.0 has removed the init_config function.
Was this necessary for the rewrite please as I was using it to create a temporary login manager on my LDAP test page so I didn't need to dip into temporarily changing the main app config.

By the way, I see you've been trying to fix the tox flow issue and I wanted to say thank you for finding the time. I had a look at one fail and thought I'd found something but the LDAP3 search just isn't returning data and I'm not experienced enough to dig into why that might be.
Thanks, Arthur

**Reply by nickw444** [reply]
This is the PR/context where it was removed: https://github.com/nickw444/flask-ldap3-login/issues/40#issuecomment-524578661

**Comment by ArthurMitchell42** [comment]
Thanks Nick. I've integrated today's release and it's looking fine so far.
My mitigation for the streamlining of init_config is to use a small helper class to temporarily change the config while the diagnostic in in progress.
 Thank you for your time looking at this.
